client/src/components/Search/index.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component } from 'react';
+import { Select, Radio, Button } from 'antd';
+import world from 'full-countries-cities';
+import 'antd/dist/antd.css';
+
+import Keywords from '../Keywords';
+
+import './style.css';
+
+const { Option } = Select;
+
+class Search extends Component {
+  state = {
+    country: undefined,
+    city: undefined,
+    lookingFor: '',
+    keyword: '',
+    viewKeywords: false,
+  };
+
+  renderOptions = list =>
+    list.map(item => (
+      <Option key={item} value={item}>
+        {item}
+      </Option>
+    ));
+
+  onSubmit = () => {};
+
+  setKeyword = value => {
+    this.setState({ keyword: value, viewKeywords: false });
+  };
+
+  hideKeyword = () => {
+    this.setState(({ viewKeywords }) => ({ viewKeywords: !viewKeywords }));
+  };
+
+  dropdownFilter = (input, option) =>
+    option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0;
+
+  render() {
+    const { country, city, keyword, viewKeywords, lookingFor } = this.state;
+
+    return (
+      <form className="search__form">
+        <div className="country-city-container">
+          <div className="autocomplete-box">
+            <div>WHICH COUNTRY?</div>
+            <Select
+              showSearch
+              placeholder="Select"
+              optionFilterProp="children"
+              onChange={value => this.setState({ country: value, city: '' })}
+              value={country}
+              filterOption={this.dropdownFilter}
+            >
+              {this.renderOptions(world.getCountryNames())}
+            </Select>
+          </div>
+          <div className="autocomplete-box">
+            <div>WHICH CITY?</div>
+            <Select
+              showSearch
+              placeholder="Select"
+              optionFilterProp="children"
+              onChange={value => {
+                this.setState({ city: value });
+              }}
+              value={city}
+              filterOption={this.dropdownFilter}
+            >
+              {country &&
+                world.getCities(country) !== undefined &&
+                this.renderOptions(world.getCities(country))}
+            </Select>
+          </div>
+        </div>
+
+        <div className="type-filter__container">
+          <Radio.Group
+            className="radio-group"
+            value={lookingFor}
+            buttonStyle="solid"
+          >
+            <p className="button-label">What Are You Looking For</p>
+            <Radio.Button
+              className="radio-button"
+              value="stay"
+              onClick={() => this.setState({ lookingFor: 'stay' })}
```

Comment:
It's easy to get into trouble passing around strings like this.

It depends how the team wants to tackle it but I think it's a good pattern to introduce some constants with valid values of `lookingFor` that you can share between components - for example here you could have something like
`onClick={() => this.setState({ lookingFor: constants.STAY })}`

Reply:
thanks, done